simpeg/utils/mesh_utils.py

```diff
@@ -94,3 +98,187 @@ def surface2inds(vrtx, trgl, mesh, boundaries=True, internal=True):
+            )
+            ZC = zc.reshape((mesh.vnC[0] * mesh.vnC[1], mesh.vnC[2]), order="F")
+            topoCC = np.zeros(ZC.shape[0])
+
+            for i in range(ZC.shape[0]):
+                if topo_cell_cutoff == "top":
+                    ind = np.argmax(ZC[i, :][ACTIND[i, :]])
+                    dz = mesh.h[2][ACTIND[i, :]][ind] * 0.5
+                elif topo_cell_cutoff == "center":
+                    dz = 0.0
+                else:
+                    raise ValueError("'topo_cell_cutoff' must be 'top' or 'center'.")
+                topoCC[i] = ZC[i, :][ACTIND[i, :]].max() + dz
+            return np.c_[mesh2D.cell_centers, topoCC]
+
+        elif mesh.dim == 2:
+            mesh1D = TensorMesh([mesh.h[0]], [mesh.x0[0]])
+            yc = mesh.cell_centers[:, 1]
+            ACTIND = active_cells.reshape((mesh.vnC[0], mesh.vnC[1]), order="F")
+            YC = yc.reshape((mesh.vnC[0], mesh.vnC[1]), order="F")
+            topoCC = np.zeros(YC.shape[0])
+            for i in range(YC.shape[0]):
+                ind = np.argmax(YC[i, :][ACTIND[i, :]])
+                if topo_cell_cutoff == "top":
+                    dy = mesh.h[1][ACTIND[i, :]][ind] * 0.5
+                elif topo_cell_cutoff == "center":
+                    dy = 0.0
+                else:
+                    raise ValueError("'topo_cell_cutoff' must be 'top' or 'center'.")
+                topoCC[i] = YC[i, :][ACTIND[i, :]].max() + dy
+            return np.c_[mesh1D.cell_centers, topoCC]
+
+    elif mesh._meshType == "TREE":
+        inds = mesh.get_boundary_cells(active_cells, direction="zu")[0]
+
+        if topo_cell_cutoff == "top":
+            dz = mesh.h_gridded[inds, -1] * 0.5
+        elif topo_cell_cutoff == "center":
+            dz = 0.0
+        return np.c_[mesh.cell_centers[inds, :-1], mesh.cell_centers[inds, -1] + dz]
+    else:
+        raise NotImplementedError(f"{type(mesh)} mesh is not supported.")
+
+
+def shift_to_discrete_topography(
+    mesh,
+    pts,
+    active_cells,
+    topo_cell_cutoff="top",
+    shift_horizontal=True,
+    heights=0.0,
+):
+    """
+    Shift locations relative to discrete surface topography.
+
+    Parameters
+    ----------
+    mesh : TensorMesh or discretize.TreeMesh
+        The mesh (2D or 3D) defining the discrete domain.
+    pts : (n, dim) numpy.ndarray
+        The original set of points being shifted relative to the discretize
+        surface topography.
+    active_cells : numpy.ndarray of int or bool, optional
+        Index array for all cells lying below the surface topography.
+    topo_cell_cutoff : {"top", "center"}
+        String to specify the cutoff for ground cells and the locations of the discrete
+        topography. For "top", ground cells lie entirely below the surface topography
+        and the discrete topography is defined on the top faces of surface cells.
+        For "center", only the cell centers must lie below the surface topograpy and
```

Comment:
Fix minor typo here:
```suggestion
        For "center", only the cell centers must lie below the surface topography and
```